monai/apps/generation/maisi/networks/diffusion_model_unet_maisi.py

```diff
@@ -0,0 +1,415 @@
+        # but time_embedding might actually be running in fp16. so we need to cast here.
+        # there might be better ways to encapsulate this.
+        t_emb = t_emb.to(dtype=x.dtype)
+        emb = self.time_embed(t_emb)
+
+        # 2. class
+        if self.num_class_embeds is not None:
+            if class_labels is None:
+                raise ValueError("class_labels should be provided when num_class_embeds > 0")
+            class_emb = self.class_embedding(class_labels)
+            class_emb = class_emb.to(dtype=x.dtype)
+            emb = emb + class_emb
+
+        # 3. input
+        if self.input_top_region_index:
+            _emb = self.top_region_index_layer(top_region_index_tensor)
+            emb = torch.cat((emb, _emb), dim=1)
+        if self.input_bottom_region_index:
+            _emb = self.bottom_region_index_layer(bottom_region_index_tensor)
+            emb = torch.cat((emb, _emb), dim=1)
+        if self.input_spacing:
+            _emb = self.spacing_layer(spacing_tensor)
+            emb = torch.cat((emb, _emb), dim=1)
+
+        # 3. initial convolution
+        h = self.conv_in(x)
+
+        # 4. down
+        if context is not None and self.with_conditioning is False:
+            raise ValueError("model should have with_conditioning = True if context is provided")
+        down_block_res_samples: list[torch.Tensor] = [h]
+        for downsample_block in self.down_blocks:
+            h, res_samples = downsample_block(hidden_states=h, temb=emb, context=context)
+            for residual in res_samples:
+                down_block_res_samples.append(residual)
+
+        # Additional residual conections for Controlnets
+        if down_block_additional_residuals is not None:
+            new_down_block_res_samples: list[torch.Tensor] = []
+            for down_block_res_sample, down_block_additional_residual in zip(
+                down_block_res_samples, down_block_additional_residuals
+            ):
+                down_block_res_sample = down_block_res_sample + down_block_additional_residual
+                new_down_block_res_samples.append(down_block_res_sample)
+
+            down_block_res_samples = new_down_block_res_samples
+
+        # 5. mid
+        h = self.middle_block(hidden_states=h, temb=emb, context=context)
+
+        # Additional residual conections for Controlnets
+        if mid_block_additional_residual is not None:
+            h = h + mid_block_additional_residual
+
+        # 6. up
+        for upsample_block in self.up_blocks:
+            res_samples = down_block_res_samples[-len(upsample_block.resnets) :]
+            down_block_res_samples = down_block_res_samples[: -len(upsample_block.resnets)]
+            h = upsample_block(hidden_states=h, res_hidden_states_list=res_samples, temb=emb, context=context)
+
+        # 7. output block
+        h = self.out(h)
+
+        if isinstance(h, torch.Tensor):
+            return h
+        return torch.tensor(h)
```

Comment:
```suggestion
        return convert_to_tensor(h)
```

Reply:
The change cannot pass `mypy` test.

Reply:
Could you please show the error info? Thanks.
Can this one help?
```
h: torch.Tensor = convert_to_tensor(self.out(h))
```

Reply:
The change cannot pass mypy test.

Reply:
I tried it at my local. You can do something like this:
Add type hint in Line 375, which is the first time define `h` and then
```
h = convert_to_tensor(self.out(h))
return h
```

Thanks!

Reply:
Thanks. I have updated this part. Please proceed.